app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
+  domain: string,
+  value: number,
+  lowestValue: number,
+  highestValue: number,
+  isShared: boolean,
+};
+
+type SingleUserTableEntry = {
+  domain: string,
+  value: number,
+};
 
 type Props = {
-  onChange: Function,
-  onCancel: Function,
+  onChange: (Array<APIUserType>) => void,
+  onCancel: () => void,
   visible: boolean,
-  selectedUserIds: Array<string>,
-  users: Array<APIUserType>,
+  selectedUsers: Array<APIUserType>,
 };
 
 type State = {
-  domain: ?string,
-  level: ?string,
+  multipleUsersEntries: Array<MultipleUserTableEntry>,
+  singleUsersEntries: Array<SingleUserTableEntry>,
+  showOnlySharedExperiences: boolean,
 };
 
 class ExperienceModalView extends React.PureComponent<Props, State> {
-  state = {
-    domain: null,
-    level: null,
-  };
-
-  increaseExperience = (): void => {
-    this.setExperience(null, true);
-  };
-
-  setExperience = (event: ?SyntheticInputEvent<>, shouldAddValue: boolean = false): void => {
-    const { domain, level } = this.state;
-    if (domain && level !== null) {
-      const newUserPromises = this.props.users.map(user => {
-        if (this.props.selectedUserIds.includes(user.id)) {
-          let newExperienceLevel = parseInt(level);
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      multipleUsersEntries: this.loadMultipleUserEntries(props.selectedUsers),
+      singleUsersEntries: this.loadSingleUserEntries(props.selectedUsers),
+      showOnlySharedExperiences: false,
+    };
+  }
 
-          if (shouldAddValue && user.experiences[domain]) {
-            newExperienceLevel = user.experiences[domain] + parseInt(level);
-          }
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.visible && !this.props.visible) {
+      this.setState({
+        multipleUsersEntries: this.loadMultipleUserEntries(nextProps.selectedUsers),
+        singleUsersEntries: this.loadSingleUserEntries(nextProps.selectedUsers),
+      });
+    }
+  }
 
-          const newUser = update(user, {
-            experiences: { [domain]: { $set: newExperienceLevel } },
+  loadMultipleUserEntries = (users: Array<APIUserType>): Array<MultipleUserTableEntry> => {
+    if (users.length <= 1) {
+      return [];
+    }
+    // find all existing experience domains
+    let sharedTableEntries: Array<MultipleUserTableEntry> = [];
+    users.forEach(user => {
+      Object.keys(user.experiences).forEach(experience => {
+        if (!sharedTableEntries.find(entry => entry.domain === experience)) {
+          sharedTableEntries.push({
+            domain: experience,
+            value: 0,
+            lowestValue: 0,
+            highestValue: -1,
+            isShared: true,
           });
-
-          return this.sendUserToServer(newUser, user);
         }
-        return Promise.resolve(user);
       });
-
-      this.closeModal(newUserPromises);
-    }
+    });
+    // create sharedTableEntries
+    sharedTableEntries = sharedTableEntries.map(entry => {
```

Comment:
Why is the variable named `sharedTableEntries` if it can contain entries with `isShared == false`? (line 95)